common/fs/src/tail.rs

```diff
@@ -79,330 +83,396 @@ pub struct Tailer {
     event_times: SyncHashMap<EntryKey, (usize, chrono::DateTime<chrono::Utc>)>,
 }
 
-impl Tailer {
-    /// Creates new instance of Tailer
-    pub fn new(
-        watched_dirs: Vec<DirPathBuf>,
-        rules: Rules,
-        lookback_config: Lookback,
-        initial_offsets: Option<HashMap<FileId, u64>>,
-    ) -> Self {
-        Self {
-            lookback_config,
-            fs_cache: Arc::new(Mutex::new(FileSystem::new(watched_dirs, rules))),
-            initial_offsets,
-            event_times: Arc::new(Mutex::new(HashMap::new())),
+fn get_file_for_path(fs: &FileSystem, next_path: &std::path::Path) -> Option<EntryKey> {
```

Comment:
All of these functions were moved from being associated with the `Tailer` struct and are now just functions.

Reply:
If anything they feel like methods on the FileSystem type, but that thing's busy enough as is...